### DISCUSSION TITLE
problem with minimal micropython firmware on stm32h723

### DISCUSSION THREAD

**Discussion by softcity003** [micropython/micropython#10178] [Core Development]
Hi All,
I want to build micropython firmware for stm32h723 , the problem is stm32h723 not supported in micropython files that's why i want to build a minimal firmware for now ,  I have followed the **ports/minimal** file and was able to build the minimal firmware .bin and .elf but when i try to install the firmware on stm32h723 with stm32 cube programmer , I got this error and the firmware not working
![Capture](https://user-images.githubusercontent.com/76036433/205892743-5415b57c-037f-474d-98a1-8e21810044d8.PNG)

Any idea about the problem?

**Comment by jahr** [comment]
Hello, I'm not sure what is the problem with Cube programmer, but AFAIK STM32H72X/73X MCUs are not supported in the stm32lib. It seems that update of the STM32H7 package to al least version 1.8.0 would be necessary (version 1.6.0 is currently used, 1.11.0 is latest).

**Reply by softcity003** [reply]
Hi jahr,

Thank you for your quick response 
Yes, I tried that, but I got several problems with building that's why I want to build a minimal firmware

**Reply by jahr** [reply]
I think that the minimal port is for STM32F4, it doesn't work on H7.

**Reply by jimmo** [reply]
That's right -- the minimal port just does the bare minimum to get a UART on an F4. You would need to update main.c to set up the H7 clock tree (much more complicated!) as well as uart_core.c to update the address of the USART1 base. Also the linker script.

**Reply by softcity003** [reply]
> Hello, I'm not sure what is the problem with Cube programmer, but AFAIK STM32H72X/73X MCUs are not supported in the stm32lib. It seems that update of the STM32H7 package to al least version 1.8.0 would be necessary (version 1.6.0 is currently used, 1.11.0 is latest).

hello again 
i did what you said i updated stm32 lib and added stm32h723 mcu files , i worked on stm32h743 file and i correct the usb pins and correct the size of flash memory but still have error with enable of usb module 
![Screenshot from 2022-12-20 13-54-39](https://user-images.githubusercontent.com/76036433/208674258-4a9afa8c-e718-4ea1-b241-f0c740e3d56b.png)
any idea ?

**Reply by jahr** [reply]
Hello, it's difficult to guess what's wrong without seeing sources ;-)
If you are not in a hurry, I advice to wait for the H72X support, see #10262

**Reply by softcity003** [reply]
> Hello, it's difficult to guess what's wrong without seeing sources ;-) If you are not in a hurry, I advice to wait for the H72X support, see #10262

Thank you

**Reply by dpgeorge** [reply]
See #10287 for initial H723 support on the stm32 port.
